### ISSUE TITLE
Kernel Panic RPI 4B

### ISSUE THREAD

**Issue by jshg46** [yrzr/opnsense-tools#20]
Anyone else had an issue with Kernel Panic on 24.1.5? New install and didn't know if this was normal.

**Comment by DFS-90** [comment]
Same problem here.

Log:
```

U-Boot 2023.10 (Feb 18 2024 - 17:59:42 +0000)

DRAM:  948 MiB (effective 7.9 GiB)
RPI 4 Model B (0xd03114)
Core:  210 devices, 16 uclasses, devicetree: board
MMC:   mmc@7e300000: 3, mmc@7e340000: 0
Loading Environment from FAT... ** Bad device specification mmc 1 **
In:    serial,usbkbd
Out:   serial,vidconsole
Err:   serial,vidconsole
Net:   eth0: ethernet@7d580000
PCIe BRCM: link up, 5.0 Gbps x1 (SSC)
starting USB...
Bus xhci_pci: Register 5000420 NbrPorts 5
Starting the controller
USB XHCI 1.00
scanning bus xhci_pci for devices... 2 USB Device(s) found
       scanning usb for storage devices... 0 Storage Device(s) found
Hit any key to stop autoboot:  0
** Booting bootflow 'mmc@7e340000.bootdev.part_1' with efi

No EFI system partition
No EFI system partition
Failed to persist EFI variables
Booting /efi\boot\bootaa64.efi
console comconsole failed to initialize
Consoles: EFI console
    Reading loader env vars from /efi/freebsd/loader.env
Setting currdev to disk0p1:
FreeBSD/arm64 EFI loader, Revision 1.1

   Command line arguments: loader.efi
   Image base: 0x39ddc000
   EFI version: 2.100
   EFI Firmware: Das U-Boot (rev 8227.4096)
   Console: efi,comconsole (0)
   Load Path: /efi\boot\bootaa64.efi
   Load Device: /VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,0000000000000000)/VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,6a00000003000000)/SD(0)/SD(1)/HD(1,MBR,0xc27c3410,0x800,0x19000)
Trying ESP: /VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,0000000000000000)/VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,6a00000003000000)/SD(0)/SD(1)/HD(1,MBR,0xc27c3410,0x800,0x19000)
Setting currdev to disk0p1:
Trying: /VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,0000000000000000)/VenHw(e61d73b9-a384-4acc-aeab-82e828f3628b,6a00000003000000)/SD(0)/SD(1)/HD(2,MBR,0xc27c3410,0x19800,0x5e6800)
Setting currdev to disk0p2:
Loading /boot/defaults/loader.conf
Loading /boot/defaults/loader.conf
Loading /boot/device.hints
Loading /boot/loader.conf
Loading /boot/loader.conf.local
cLoading kernel...
/boot/kernel/kernel text=0x2a8 text=0x9429e0 text=0x238bbc data=0x1b8538 data=0x0+0x2b2000 0x8+0x13ba68+0x8+0x161c23\
Loading configured modules...
umass...ukbd...usb...carp.../boot/kernel/carp.ko text=0x3500 text=0x6f80 data=0xcd0+0x48 0x8+0x19f8+0x8+0x12e9
pfsync.../boot/kernel/pfsync.ko text=0x2f63 text=0x7fcc data=0xb38+0x8 0x8+0x1848+0x8+0x1198
loading required module 'pf'
/boot/kernel/pf.ko text=0xf7e4 text=0x405dc data=0x5d20+0x43c 0x8+0x62a0+0x8+0x5171
/boot/entropy.../boot/entropy size=0x1000
pflog.../boot/kernel/pflog.ko text=0xf78 text=0x8a8 data=0x430 0x8+0xa08+0x8+0x718
ugen...can't find 'ugen'
failed!
if_vlan...if_tap...if_enc.../boot/kernel/if_enc.ko text=0x15ca text=0x934 data=0x750 0x8+0xca8+0x8+0xb71
if_bridge.../boot/kernel/if_bridge.ko text=0x3884 text=0x6e2c data=0xde8+0x8 0x8+0x1c80+0x8+0x16ac
loading required module 'bridgestp'
/boot/kernel/bridgestp.ko text=0x12e5 text=0x4cd8 data=0x2c8+0x28 0x8+0xb28+0x8+0x6af
if_lagg.../boot/kernel/if_lagg.ko text=0x3d0f text=0x9890 data=0xd10+0x8 0x8+0x1c50+0x8+0x159d
loading required module 'if_infiniband'
/boot/kernel/if_infiniband.ko text=0x1102 text=0x112c data=0x300+0x8 0x8+0x8a0+0x8+0x590
if_gre.../boot/kernel/if_gre.ko text=0x2736 text=0x40f8 data=0x8c8+0x40 0x8+0x1710+0x8+0xfd9
if_gif.../etc/hostid...can't find '/etc/hostid'
failed!
uhid.../boot/kernel/uhid.ko text=0x1fe0 text=0x1650 data=0x6d0+0x10 0x8+0xeb8+0x8+0xa75
if_tun...pf...
Hit [Enter] to boot immediately, or any other key for command prompt.
Booting [/boot/kernel/kernel]...
Using DTB provided by EFI at 0x39eff000.
EFI framebuffer information:
addr, size     0x3eaf5000, 0x103000
dimensions     592 x 448
stride         592
masks          0x00ff0000, 0x0000ff00, 0x000000ff, 0xff000000
---<<BOOT>>---
KDB: debugger backends: ddb
KDB: current backend: ddb
WARNING: Cannot find freebsd,dts-version property, cannot check DTB compliance
Copyright (c) 1992-2021 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
        The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 13.2-RELEASE-p11 stable/24.1-n255007-1d6e165fb40 SMP arm64
FreeBSD clang version 14.0.5 (https://github.com/llvm/llvm-project.git llvmorg-14.0.5-0-gc12386ae247c)
VT(efifb): resolution 592x448
module firmware already present!
real memory  = 8443125760 (8051 MB)
avail memory = 8207859712 (7827 MB)
Starting CPU 1 (1)
Starting CPU 2 (2)
Starting CPU 3 (3)
FreeBSD/SMP: Multiprocessor System Detected: 4 CPUs
random: unblocking device.
random: entropy device external interface
MAP 39f11000 mode 2 pages 1
MAP 39f15000 mode 2 pages 1
MAP 39f17000 mode 2 pages 18
MAP 3b340000 mode 2 pages 16
MAP fe100000 mode 0 pages 1
kbd0 at kbdmux0
ofwbus0: <Open Firmware Device Tree>
simplebus0: <Flattened device tree simple bus> on ofwbus0
ofw_clkbus0: <OFW clocks bus> on ofwbus0
clk_fixed0: <Fixed clock> on ofw_clkbus0
clk_fixed1: <Fixed clock> on ofw_clkbus0
clk_fixed2: <Fixed clock> on ofwbus0
clk_fixed3: <Fixed clock> on ofwbus0
simplebus1: <Flattened device tree simple bus> on ofwbus0
simplebus2: <Flattened device tree simple bus> on ofwbus0
regfix0: <Fixed Regulator> on ofwbus0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
regfix1: <Fixed Regulator> on ofwbus0
regfix2: <Fixed Regulator> on ofwbus0
regfix3: <Fixed Regulator> on ofwbus0
regfix4: <Fixed Regulator> on ofwbus0
simplebus3: <Flattened device tree simple bus> on ofwbus0
simple_mfd0: <Simple MFD (Multi-Functions Device)> mem 0x7d5d2000-0x7d5d2eff on simplebus0
bcm2835_firmware0: <BCM2835 Firmware> on simplebus0
ofw_clkbus1: <OFW clocks bus> on bcm2835_firmware0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
psci0: <ARM Power State Co-ordination Interface Driver> on ofwbus0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
gic0: <ARM Generic Interrupt Controller> mem 0x40041000-0x40041fff,0x40042000-0x40043fff,0x40044000-0x40045fff,0x40046000-0x40047fff irq 30 on simplebus0
gic0: pn 0x2, arch 0x2, rev 0x1, implementer 0x43b irqs 256
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
gpio0: <BCM2708/2835 GPIO controller> mem 0x7e200000-0x7e2000b3 irq 14,15 on simplebus0
gpiobus0: <OFW GPIO bus> on gpio0
gpio1: <Raspberry Pi Firmware GPIO controller> on bcm2835_firmware0
gpiobus1: <GPIO bus> on gpio1
regfix0: Cannot configure GPIO pin: 5
REGNODE_INIT failed: 6
regfix0: Cannot register regulator.
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
regfix2: Cannot set GPIO pin: 6
REGNODE_INIT failed: 6
regfix2: Cannot register regulator.
mbox0: <BCM2835 VideoCore Mailbox> mem 0x7e00b880-0x7e00b8bf irq 13 on simplebus0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
gpioregulator0: <GPIO controlled regulator> on ofwbus0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
generic_timer0: <ARMv8 Generic Timer> irq 4,5,6,7 on ofwbus0
Timecounter "ARM MPCore Timecounter" frequency 54000000 Hz quality 1000
Event timer "ARM MPCore Eventtimer" frequency 54000000 Hz quality 1000
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
usb_nop_xceiv0: <USB NOP PHY> on ofwbus0
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
clk_fixed4: <Fixed clock> disabled on ofwbus0
clk_fixed4: Cannot FDT parameters.
device_attach: clk_fixed4 attach returned 6
bcm2835_clkman0: <BCM283x Clock Manager> mem 0x7e101000-0x7e102fff on simplebus0
gpioc0: <GPIO controller> on gpio0
uart0: <PrimeCell UART (PL011)> mem 0x7e201000-0x7e2011ff irq 16 on simplebus0
uart0: console (115200,n,8,1)
sdhci_bcm0: <Broadcom 2708 SDHCI controller> mem 0x7e300000-0x7e3000ff irq 24 on simplebus0
Fatal data abort:
  x0:         ffffffff
  x1:                0
  x2: ffff0000009446ff (cam_status_table + 14f7)
  x3:               6e
  x4: ffff000000fe760c (initstack + 360c)
  x5:               6e
  x6: ffff00000019c940 (simplebus_get_property + 0)
  x7: 6d63625f69636864
  x8: ffff000000ed89e0 (thread0_st + 0)
  x9:               20
 x10:                0
 x11:                1
 x12:  300000000006e65
 x13:   fefefefeff0100
 x14:              823
 x15:               1a
 x16:                0
 x17:                0
 x18: ffff000000fe77e0 (initstack + 37e0)
 x19:         ffffffff
 x20:                0
 x21: ffff000000c83000 (pagertab + 60)
 x22: ffff000000c83000 (pagertab + 60)
 x23: ffffa00000faa838
 x24: ffff0000009b038e (digits + 4627)
 x25: ffff0000009de4be (console_pausestr + e4d5)
 x26: ffff00000096b67a (cam_status_table + 28472)
 x27: ffffa00000f76140
 x28:         31e00000
 x29: ffff000000fe77e0 (initstack + 37e0)
  sp: ffff000000fe77e0
  lr: ffff0000008ee58c (bcm_sdhci_attach + 31c)
 elr: ffff0000008e8684 (bcm_dma_allocate + 88)
spsr:         a00000c5
 far:               20
 esr:         96000004
panic: vm_fault failed: ffff0000008e8684
cpuid = 0
time = 1
KDB: stack backtrace:
db_trace_self() at db_trace_self_wrapper+0x30
         pc = 0xffff00000083d914  lr = 0xffff0000000f7aa8
         sp = 0xffff000000fe7230  fp = 0xffff000000fe7430

db_trace_self_wrapper() at vpanic+0x13c
         pc = 0xffff0000000f7aa8  lr = 0xffff0000004f5a0c
         sp = 0xffff000000fe7440  fp = 0xffff000000fe74a0

vpanic() at panic+0x44
         pc = 0xffff0000004f5a0c  lr = 0xffff0000004f58cc
         sp = 0xffff000000fe74b0  fp = 0xffff000000fe7560

panic() at data_abort+0x240
         pc = 0xffff0000004f58cc  lr = 0xffff0000008618e8
         sp = 0xffff000000fe7570  fp = 0xffff000000fe75f0

data_abort() at handle_el1h_sync+0x10
         pc = 0xffff0000008618e8  lr = 0xffff000000840010
         sp = 0xffff000000fe7600  fp = 0xffff000000fe7750

handle_el1h_sync() at bcm_sdhci_attach+0x318
         pc = 0xffff000000840010  lr = 0xffff0000008ee588
         sp = 0xffff000000fe7760  fp = 0xffff000000fe77e0

bcm_sdhci_attach() at bcm_sdhci_attach+0x318
         pc = 0xffff0000008ee588  lr = 0xffff0000008ee588
         sp = 0xffff000000fe77f0  fp = 0xffff000000fe7850

bcm_sdhci_attach() at device_attach+0x3fc
         pc = 0xffff0000008ee588  lr = 0xffff0000005378bc
         sp = 0xffff000000fe7860  fp = 0xffff000000fe78b0

device_attach() at bus_generic_new_pass+0x120
         pc = 0xffff0000005378bc  lr = 0xffff000000539a5c
         sp = 0xffff000000fe78c0  fp = 0xffff000000fe7910

bus_generic_new_pass() at bus_generic_new_pass+0xb0
         pc = 0xffff000000539a5c  lr = 0xffff0000005399ec
         sp = 0xffff000000fe7920  fp = 0xffff000000fe7950

bus_generic_new_pass() at bus_generic_new_pass+0xb0
         pc = 0xffff0000005399ec  lr = 0xffff0000005399ec
         sp = 0xffff000000fe7960  fp = 0xffff000000fe7990

bus_generic_new_pass() at bus_generic_new_pass+0xb0
         pc = 0xffff0000005399ec  lr = 0xffff0000005399ec
         sp = 0xffff000000fe79a0  fp = 0xffff000000fe79d0

bus_generic_new_pass() at root_bus_configure+0x40
         pc = 0xffff0000005399ec  lr = 0xffff00000053bb28
         sp = 0xffff000000fe79e0  fp = 0xffff000000fe7a10

root_bus_configure() at mi_startup+0x11c
         pc = 0xffff00000053bb28  lr = 0xffff000000469ad4
         sp = 0xffff000000fe7a20  fp = 0xffff000000fe7a50

mi_startup() at virtdone+0x78
         pc = 0xffff000000469ad4  lr = 0xffff0000000008b4
         sp = 0xffff000000fe7a60  fp = 0x0000000000000000

KDB: enter: panic
[ thread pid 0 tid 100000 ]
Stopped at      kdb_enter+0x48: undefined       f906011f
db>
db> clk_fixed4: Cannot FDT parameters.
No such command; use "help" to list available commands
db>
db> clk_fixed4: Cannot FDT parameters.
No such command; use "help" to list available commands
db> panic: vm_fault failed: ffff0000008e8684

```

**Comment by DFS-90** [comment]
As mentioned [here](https://github.com/yrzr/opnsense-tools/issues/6), replacing all files in MSDOSBOOT partition with the ones from the FreeBSD image solves the problem.

I found out that replacing them with the ones from FreeBSD 14.0 leads to kernel panic.
Replacing them with those from FreeBSD 13.2 works flawlessly.

Please find the mentioned files attached:
[FreeBSD-13.2_0.fat.zip](https://github.com/user-attachments/files/15753385/FreeBSD-13.2_0.fat.zip)

Source:
[FreeBSD 13.2 RPI image](https://download.freebsd.org/releases/arm64/aarch64/ISO-IMAGES/13.2/FreeBSD-13.2-RELEASE-arm64-aarch64-RPI.img.xz)

Best regards,
David

**Comment by gdman07** [comment]
I've tried replacing all the files, but this causes the pi4 to boot loop.  Could you provide instructions on which files you moved?  I get an error 110 and the pi just reboots

**Comment by DFS-90** [comment]
I replaced all files on MSDOSBOOT partition with those from the "0.fat" file:
![grafik](https://github.com/yrzr/opnsense-tools/assets/87578070/877504bf-66d3-48a5-813d-a58eaa6bb26e)

Check your config.txt - mine looks like this:
```
[all]
arm_64bit=1
dtparam=audio=on,i2c_arm=on,spi=on
dtoverlay=mmc
dtoverlay=disable-bt
device_tree_address=0x4000
kernel=u-boot.bin
enable_uart=1
uart_2ndstage=1

[pi4]
hdmi_safe=1
armstub=armstub8-gic.bin
```

**Comment by gdman07** [comment]
Thank you DFS-90!  My issue was, I used the files from the wrong freebsd.  Thanks for you help

**Comment by 459below** [comment]
Thanks @DFS-90 and everyone else

I was affected by it on the RPI 3 and RPI 4, too. I guess this is an RPI 3, RPI 4, FreeBSD 14 upstream, regression and UBOOT thing.

In any case using the FreeBSD 13 Boot partition worked for both RPI 3 and RPI 4 to fix booting of 24.1.5. It also worked for my locally build version of 24.1.6.

![Screenshot from 2024-06-26 14-42-46](https://github.com/yrzr/opnsense-tools/assets/11325631/6ec05eb5-760e-4617-a465-56e806b1e218)

As of right now I am building 24.1.9 on my RPI 5. This usually took more than 24 hrs of pure compile time. I am looking forward to build 24.7, as soon as it gets the "Stable"-sticker.

Slightly offtopic about CPU speed:
```
# check current cpu speed
root@OPNsense:~ # sysctl dev.cpu.0.freq
dev.cpu.0.freq: 600
root@OPNsense:~ # sysrc powerd_enable="YES"
root@OPNsense:~ # /etc/rc.d/powerd restart
root@OPNsense:~ # sysctl dev.cpu.0.freq
dev.cpu.0.freq: 1500
```
Would enable the CPU to go from the minimum 600 to the normal spec CPU speed of 1500 on-demand. Instantly doubled my download speed.

**Comment by 459below** [comment]
I just ran my compile of 24.7.3, which of course is now on FreeBSD 14.1. Both the build environment as well as the system.

The Kernel Panic issue did not occurr  to me on an entirely fresh build FreeBSD 14.1 build system with no modifications made, except for the part that was needed before, too.:
```
cd /usr/ports/ports-mgmt/pkg
make package
make reinstall
```

---

So I think the issue is now fixed. If somebody wants to reproduce, we could definitely close.

![image](https://github.com/user-attachments/assets/f804bcad-cf8f-4f98-a422-eb5307379e47)
